clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy

```diff
@@ -204,16 +210,48 @@ class BasicGoogleDeployHandler implements DeployHandler<BasicGoogleDeployDescrip
       : null
 
     def migCreateOperation
+    def instanceGroupManager = new InstanceGroupManager()
+        .setName(serverGroupName)
+        .setBaseInstanceName(serverGroupName)
+        .setInstanceTemplate(instanceTemplateUrl)
+        .setTargetSize(description.targetSize)
+        .setTargetPools(targetPools)
+        .setAutoHealingPolicies(autoHealingPolicy)
+
+    def hasBackendServices = description.instanceMetadata &&
+        description.instanceMetadata.containsKey(GoogleServerGroup.View.BACKEND_SERVICE_NAMES)
+
+    if (hasBackendServices && (description?.loadBalancingPolicy || description?.source?.serverGroupName))  {
+      NamedPort namedPort = null
+      def sourceGroupName = description?.source?.serverGroupName
+      // Note: this favors the explicitly specified load balancing policy over the source server group.
+      if (sourceGroupName && !description?.loadBalancingPolicy) {
+        def sourceServerGroup = googleClusterProvider.getServerGroup(description.accountName, description.source.region, sourceGroupName)
+        if (!sourceServerGroup) {
+          log.warn("Could not locate source server group ${sourceGroupName} to update named port.")
+        }
+        namedPort = new NamedPort(
+            name: GoogleHttpLoadBalancingPolicy.HTTP_PORT_NAME,
```

Comment:
Can't the creator of the load balancer choose a port other than 'http'? If so, don't we need to get the name from the load balancer's backend service?